napari/resources/build_icons.py

```diff
@@ -73,20 +84,64 @@ def build_icons():
     with open(qrcpath, 'w') as f:
         f.write(qrc_string)
 
-
-if __name__ == "__main__":
-    from subprocess import run
-
-    qtpy_file = join(resources_dir, 'qt.py')
-
-    build_icons()
+    return qrcpath
+
+
+def build_pyqt_resources(
+    out_path: Optional[str] = None, overwrite: bool = False
+) -> str:
+    """Build a res.qrc file from icons and convert for python usage.
+
+    calls :func:`build_resources` and then converts using ``pyside2-rcc`` or
+    ``pyrcc5`` depending on which is installed in the environment.
+    Finally, cleans up autogenerated icon.svgs and res.qrc file after _qt.py
+    file is generated
+
+    References
+    ----------
+    https://doc.qt.io/qt-5/resources.html
+
+    Parameters
+    ----------
+    out_path : str, optional
+        Path to write the converted "qt.py" resource file, by default
+        os.path.join(RESOURCES_DIR, 'qt.py')
+    overwrite : bool, optional
+        Whether to force rebuilding of the out file, by default False
+
+    Returns
+    -------
+    str
+        Path to the converted python resource file.  Import this file to make
+        the svg and other resources available to Qt stylesheets.
+    """
+    out_path = out_path or os.path.join(RESOURCES_DIR, '_qt.py')
+    if os.path.exists(out_path) and not overwrite:
+        return out_path
+
+    # build the resource file to the same path
+    qrc_path = os.path.join(os.path.dirname(out_path), 'res.qrc')
+    qrc_path = build_resources(qrc_path, overwrite=overwrite)
+
+    # then convert it to a python file
     try:
-        run(['pyside2-rcc', '-o', qtpy_file, qrcpath])
+        run(['pyrcc5', '-o', out_path, qrc_path])
     except FileNotFoundError:
-        run(['pyrcc5', '-o', qtpy_file, qrcpath])
+        run(['pyside2-rcc', '-o', out_path, qrc_path])
 
-    with open(qtpy_file, "rt") as fin:
+    # make sure we import from qtpy
+    with open(out_path, "rt") as fin:
         data = fin.read()
         data = data.replace('PySide2', 'qtpy').replace('PyQt5', 'qtpy')
-    with open(qtpy_file, "wt") as fin:
+    with open(out_path, "wt") as fin:
         fin.write(data)
+
+    # cleanup.
+    # we do this here because pip uninstall napari would not collect these
```

Comment:
I'm a little confused... Does this mean this would need regenerating on each boot? Or are these files used only once during creation of the .py, and then not looked at again?

Reply:
the latter, and that was something I learned here too.  The file generated by `pyrcc5` or `pyside22-rcc` is actually a complete  bytes file with all necessary info ... neither the themed SVGs nor the res.qrc are necessary after creating (what we called) `qt.py`.

Reply:
:+1: !